src/backend/distributed/utils/shard_utils.c

```diff
@@ -55,3 +59,52 @@ GetLongestShardName(Oid citusTableOid, char *finalRelationName)
 
 	return longestShardName;
 }
+
+
+/*
+ * GetLongestHypotheticalShardName is a utility function that creates a hypothetical shard
+ * name for a table that is not distributed yet.
+ */
+char *
+GetLongestHypotheticalShardName(Oid parentTableOid, char *relationName)
```

Comment:
It is kind of hard to follow the naming. The function name is very generic but:

1) We name the function as `GetLongestHypotheticalShardName` implying that we are calling this on a local table. However, the function assumes that the input `parentTableOid` is a distributed table
2) The function parameter is `parentTableOid`, but the function name seems to operate on all kinds of tables

In general, I was thinking that we should probably extract `pg_sequence_last_value` related parts to a different function, and the calculations regarding to parent -> child relations in a different function. This function seems to do many things at once, and slightly implicitly

Reply:
1. How does `GetLongestShardNameForLocalPartition` sound? I renamed it.
2. Would you want me to add an assertion or some error messages that should never occur here? I could not find a better alternative for this.

I also moved `pg_sequence_last_value` calls in a separate helper function